core/client/fs/src/main/java/alluxio/client/file/FileSystemContextReinitializer.java

```diff
@@ -55,14 +62,17 @@
   public FileSystemContextReinitializer(FileSystemContext context) {
     mContext = context;
     mExecutor = new ConfigHashSync(context);
-    mExecutorService = Executors.newSingleThreadExecutor(ThreadFactoryUtils.build(
-        "config-hash-master-heartbeat-%d", true));
-    mExecutorService.submit(
-        new HeartbeatThread(HeartbeatContext.META_MASTER_CONFIG_HASH_SYNC, mContext.getId(),
-            mExecutor, mContext.getClientContext().getClusterConf().getMs(
-                PropertyKey.USER_CONF_SYNC_INTERVAL),
-            mContext.getClientContext().getClusterConf(),
-            mContext.getClientContext().getUserState()));
+    REINIT_EXECUTOR.scheduleAtFixedRate(() -> {
```

Comment:
Can you store the Future<?> in order to cancel it during reinitializer shutdown?